markets/perps-market/contracts/storage/AsyncOrder.sol

```diff
@@ -151,6 +151,25 @@ library AsyncOrder {
         checkWithinSettlementWindow(order, strategy);
     }
 
+    /**
+     * @dev Updates the order with the new commitment request data and settlement time.
+     * @dev Reverts if there's a pending order.
+     * @dev Reverts if accont cannot open a new position (due to max allowed reached).
+     */
+    function updateValid(
+        Data storage self,
+        OrderCommitmentRequest memory newRequest,
+        SettlementStrategy.Data storage strategy
+    ) internal {
+        checkPendingOrder(newRequest.accountId);
+
+        PerpsAccount.load(newRequest.accountId).validateMaxPositions(newRequest.marketId);
```

Comment:
can just do the account loading in the `validateMaxPositions` function.

so could just do `PerpsAccount.validateMaxPositions(newRequest.accountId, newRequest.marketId);`

non blocking

Reply:
good call, will update